### PR TITLE
IRSA-641 Extend the saving/uploading feature to include the choice of…

### PR DESCRIPTION
**Pull request by wmiipac**
I have reverted most of the changes in FileUpload and FileUploadViewPanel (these were written for the current image search specifically). So it seems to be incomplete.
for the image select upload and time series viewer upload panel, the upload workspace needs to be connected to the file picker. There is no selection logic applied for now.
To view the extended save/upload panel:
1. run an image search,  from image toolbar, click save icon
2.    in firefly/image search panel
3.    in firefly/catalog search load catalog panel
4.   in firefly/Upload panel
5.   time series viewer
6.   added a File Upload dialog in firefly-dev.html page, for testing development of file pick only

### CODE REVIEW

File: src/firefly/js/core/ReduxFlux.js [hunk 1]
```diff
@@ -160,6 +163,16 @@ actionCreators.set('exampleDialog', (rawAction) => {
     return rawAction;
 });
 
+actionCreators.set('fileuploadDialog', (rawAction) => {
+    showFileUploadDialog();
+    return rawAction;
+});
+
+actionCreators.set('downloadDialog', (rawAction) => {
+    DownloadOptionPanel();
```
Comment (loitly) [inline src/firefly/js/core/ReduxFlux.js]: showExampleDialog was done early on and was not cleaned up properly.  You should not have to register these dialogs as an action creator.  The action is SHOW_DIALOG in ComponentCntlr.  The added lines in this file should be removed.

File: src/firefly/js/ui/FileUpload.jsx [hunk 2]
```diff
@@ -13,9 +13,9 @@ const UL_URL = `${getRootURL()}sticky/CmdSrv?${ServerParams.COMMAND}=${ServerPar
 
 
 function FileUploadView({fileType, isLoading, label, valid, wrapperStyle,  message, onChange, value, labelWidth,
-                         innerStyle, isFromURL, onUrlAnalysis, fileNameStyle}) {
-    var style = !isFromURL ? Object.assign({color: 'transparent', border: 'none', background: 'none'}, innerStyle) : (innerStyle || {});
-    var fileName = (!isFromURL && value) ?  value.split(/(\\|\/)/g).pop() : 'No file chosen';
+                         innerStyle, isFromURL, isFromWsURL, isFromFile, uploadSrc, onUrlAnalysis, fileNameStyle}) {
```
Comment (loitly) [inline src/firefly/js/ui/FileUpload.jsx]: uploadSrc added but not used.

**Comment by ejoliet** [comment]
Another question:
- Is there also a download/upload demo html page using API where i can trigger the dialog option for choosing local or workspace files? That would definitively be helpful.

**Comment by cwang2016** [comment]
file upload from local file system and from url on both firefly/image search panel and  firefly/catalog search load catalog panel have some running issues which is mainly caused by:
- there are two props, isFromFile and  isFromWsURL, added to 'FileUpload' component in order to denote if the upload is from file, url or workspace, these newly defined props should be added onto 'FileUpload' instance used by 'FITS file' under image search and 'Load Catalog' under catalog search.

File: src/firefly/js/ui/FileUpload.jsx [hunk 3]
```diff
@@ -139,7 +163,7 @@ function handleChange(ev, fireValueChange, type, fileAnalysis) {
     });
 }
 
-function makeDoUpload(file, type, isFromURL, fileAnalysis) {
+function makeDoUpload(file, type, isFromURL, isFromWsURL, fileAnalysis) {
```
Comment (cwang2016) [inline src/firefly/js/ui/FileUpload.jsx]: parameter 'isFromWsURL' is introduced into this function 'makeDoUpload', please check if every function call on 'makeDoUpload' in this file has add this newly added parameter.

**Pull request closed without merging**